File: backend/app/schemas/telegram.py
"""Pydantic schemas for /api/v1/telegram/*."""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, ConfigDict


class _BaseModel(BaseModel):
    model_config = ConfigDict(from_attributes=True, str_strip_whitespace=True)


class TelegramBindingStatus(_BaseModel):
    bound: bool
    telegram_user_id: int | None = None
    telegram_username: str | None = None
    bound_at: datetime | None = None
    notifications_enabled: bool = True
    notify_medications: bool = True
    notify_visits: bool = True
    notify_daily_summary: bool = False
    notify_health_tips: bool = False


class TelegramBindingCodeResponse(_BaseModel):
    code: str
    expires_at: datetime
    bot_username: str
    deep_link: str


class TelegramNotificationSettingsUpdate(_BaseModel):
    notifications_enabled: bool | None = None
    notify_medications: bool | None = None
    notify_visits: bool | None = None
    notify_daily_summary: bool | None = None
    notify_health_tips: bool | None = None


class TelegramTestMessageResponse(_BaseModel):
    delivered: bool

File: frontend/src/api/telegram.ts
import apiClient from './client'

export interface TelegramBindingStatus {
  bound: boolean
  telegram_user_id: number | null
  telegram_username: string | null
  bound_at: string | null
  notifications_enabled: boolean
  notify_medications: boolean
  notify_visits: boolean
  notify_daily_summary: boolean
  notify_health_tips: boolean
}

export interface TelegramBindingCode {
  code: string
  expires_at: string
  bot_username: string
  deep_link: string
}

export interface TelegramNotificationsUpdate {
  notifications_enabled?: boolean
  notify_medications?: boolean
  notify_visits?: boolean
  notify_daily_summary?: boolean
  notify_health_tips?: boolean
}

export const telegramApi = {
  getBinding() {
    return apiClient.get<TelegramBindingStatus>('/v1/telegram/binding')
  },
  generateCode() {
    return apiClient.post<TelegramBindingCode>('/v1/telegram/binding/code')
  },
  unbind() {
    return apiClient.delete<void>('/v1/telegram/binding')
  },
  updateNotifications(payload: TelegramNotificationsUpdate) {
    return apiClient.patch<TelegramBindingStatus>(
      '/v1/telegram/binding/notifications',
      payload,
    )
  },
  sendTest() {
    return apiClient.post<{ delivered: boolean }>('/v1/telegram/binding/test')
  },
}
